src/queue.rs

```diff
@@ -112,12 +112,21 @@ impl<M: GuestAddressSpace> DescriptorChain<M> {
             return None;
         }
 
-        let desc_table_size = size_of::<Descriptor>() * queue_size as usize;
-        let slice = mem.get_slice(desc_table, desc_table_size).ok()?;
```

Comment:
I agree that we should have a testcase if there is a bug here.

Reply:
The issue is that, when the whole descriptor table crosses region boundary, it's  legal and should work. With current implementation it fails because the get_slice() works for a single region.

Reply:
@sameo You're confusing `GuestRegion::checked_offset` with `GuestMemory::checked_offset`. But we need a testcase.

Reply:
Add one test case to cover this case:)